sqlalchemy-stubs/sql/compiler.pyi

```diff
@@ -185,73 +215,86 @@ class SQLCompiler(Compiled):
-    def visit_collation(self, element: Any, **kw: Any): ...
-    def visit_fromclause(self, fromclause: Any, **kwargs: Any): ...
-    def visit_index(self, index: Any, **kwargs: Any): ...
-    def visit_typeclause(self, typeclause: Any, **kw: Any): ...
-    def post_process_text(self, text: Any): ...
-    def escape_literal_column(self, text: Any): ...
+    ) -> _VisitResult: ...
+    def visit_collation(self, element: Any, **kw: Any) -> _VisitResult: ...
+    def visit_fromclause(
+        self, fromclause: Any, **kwargs: Any
+    ) -> _VisitResult: ...
+    def visit_index(self, index: Any, **kwargs: Any) -> _VisitResult: ...
+    def visit_typeclause(self, typeclause: Any, **kw: Any) -> _VisitResult: ...
+    def post_process_text(self, text: Any) -> _VisitResult: ...
+    def escape_literal_column(self, text: Any) -> _VisitResult: ...
     def visit_textclause(
         self,
         textclause: Any,
         add_to_result_map: Optional[Any] = ...,
         **kw: Any,
-    ): ...
+    ) -> _VisitResult: ...
     def visit_textual_select(
         self,
         taf: Any,
         compound_index: Optional[Any] = ...,
         asfrom: bool = ...,
         **kw: Any,
-    ): ...
-    def visit_null(self, expr: Any, **kw: Any): ...
-    def visit_true(self, expr: Any, **kw: Any): ...
-    def visit_false(self, expr: Any, **kw: Any): ...
-    def visit_tuple(self, clauselist: Any, **kw: Any): ...
-    def visit_clauselist(self, clauselist: Any, **kw: Any): ...
-    def visit_case(self, clause: Any, **kwargs: Any): ...
-    def visit_type_coerce(self, type_coerce: Any, **kw: Any): ...
-    def visit_cast(self, cast: Any, **kwargs: Any): ...
-    def visit_over(self, over: Any, **kwargs: Any): ...
-    def visit_withingroup(self, withingroup: Any, **kwargs: Any): ...
-    def visit_funcfilter(self, funcfilter: Any, **kwargs: Any): ...
-    def visit_extract(self, extract: Any, **kwargs: Any): ...
-    def visit_scalar_function_column(self, element: Any, **kw: Any): ...
+    ) -> _VisitResult: ...
+    def visit_null(self, expr: Any, **kw: Any) -> _VisitResult: ...
+    def visit_true(self, expr: Any, **kw: Any) -> _VisitResult: ...
+    def visit_false(self, expr: Any, **kw: Any) -> _VisitResult: ...
+    def visit_tuple(self, clauselist: Any, **kw: Any) -> _VisitResult: ...
+    def visit_clauselist(self, clauselist: Any, **kw: Any) -> _VisitResult: ...
+    def visit_case(self, clause: Any, **kwargs: Any) -> _VisitResult: ...
+    def visit_type_coerce(
+        self, type_coerce: Any, **kw: Any
+    ) -> _VisitResult: ...
+    def visit_cast(self, cast: Any, **kwargs: Any) -> _VisitResult: ...
+    def visit_over(self, over: Any, **kwargs: Any) -> _VisitResult: ...
+    def visit_withingroup(
+        self, withingroup: Any, **kwargs: Any
+    ) -> _VisitResult: ...
+    def visit_funcfilter(
+        self, funcfilter: Any, **kwargs: Any
+    ) -> _VisitResult: ...
+    def visit_extract(self, extract: Any, **kwargs: Any) -> _VisitResult: ...
```

Comment:
for all these `visit_*` the input elements could be typed, but I don't think we would have much gain from it.

for sure it would be a future nice to have but this is more than enough :)